### PR TITLE
[MIG] Module website_sale_charge_payment_fee: Migrating from v10 to v11

### PR DESCRIPTION
**Pull request by aparragithub**

### CODE REVIEW

File: website_sale_charge_payment_fee/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+{
+    "name": "eCommerce: charge payment fee",
+    "summary": "Payment fee charged to customer",
+    "version": "11.0.1.0.0",
+    "development_status": "Beta",
+    "category": "Website",
+    "website": "https://github.com/OCA/e-commerce",
+    "author": "Agile Business Group, AITIC S.A.S, "
```
Comment (mamcode) [inline website_sale_charge_payment_fee/__manifest__.py]: You cannot modify the authors of the module without first checking with the original authors (if you are adding any new functionality to the module), if it is not the case, you just need to add yourself to the list of contributors in readme/CONTRIBUTORS.rst.

File: website_sale_charge_payment_fee/templates/website_sale.xml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            <t t-if="line.payment_fee_line">
+                <strong t-esc="line.product_id.with_context(display_default_code=False).display_name"/>
+            </t>
+            <t t-if="not line.payment_fee_line">
+                <a t-att-href="line.product_id.website_url">
+                    <t t-raw="0"/>
+                </a>
+            </t>
+        </a>
+    </template>
```
Comment (mamcode) [inline website_sale_charge_payment_fee/templates/website_sale.xml]: it seems that website_sale is complaining in its tests about this template... Please try with the following change:
```suggestion
    <template id="cart_line_product_link_fee" inherit_id="website_sale.cart_lines">
        <xpath expr="//a[@t-attf-href='/shop/product/#{ slug(line.product_id.product_tmpl_id) }']" position="replace">
            <t t-if="line.payment_fee_line">
                <strong t-esc="line.product_id.with_context(display_default_code=False).display_name"/>
            </t>
            <t t-if="not line.payment_fee_line">
                <a t-attf-href="/shop/product/#{ slug(line.product_id.product_tmpl_id) }">
                    <strong t-esc="line.product_id.with_context(display_default_code=False).display_name" />
                </a>
            </t>
        </xpath>
    </template>
```
Run the tests to verify that everything is ok ...
Also consider including this xpath in the `cart_lines_fee template`, so you don't have two templates inheriting from `website_sale.cart_lines`.

**Review by mamcode — CHANGES_REQUESTED** [review]
@aparragithub Thanks for this migration, functionally the module is fine but there are some points to improve... Look my comments please.

**Comment by mamcode** [comment]
@aparragithub Also please try to improve the following xpath expressions to avoid warnings
```
2020-07-24 14:38:45,909 10307 WARNING oca-ecommerce-unit-tests-11.0-2 odoo.addons.base.ir.ir_ui_view: Error-prone use of @class in view payment_tokens_list_fee (website_sale_charge_payment_fee.payment_tokens_list_fee): use the hasclass(*classes) function to filter elements by their classes
2020-07-24 14:38:45,954 10307 WARNING oca-ecommerce-unit-tests-11.0-2 odoo.addons.base.ir.ir_ui_view: Error-prone use of @class in view cart_lines_fee (website_sale_charge_payment_fee.cart_lines_fee): use the hasclass(*classes) function to filter elements by their classes
2020-07-24 14:38:45,954 10307 WARNING oca-ecommerce-unit-tests-11.0-2 odoo.addons.base.ir.ir_ui_view: Error-prone use of @class in view cart_lines_fee (website_sale_charge_payment_fee.cart_lines_fee): use the hasclass(*classes) function to filter elements by their classes
2020-07-24 14:38:46,024 10307 WARNING oca-ecommerce-unit-tests-11.0-2 odoo.addons.base.ir.ir_ui_view: Error-prone use of @class in view payment_tokens_list_fee (website_sale_charge_payment_fee.payment_tokens_list_fee): use the hasclass(*classes) function to filter elements by their classes
2020-07-24 14:38:46,032 10307 WARNING oca-ecommerce-unit-tests-11.0-2 odoo.addons.base.ir.ir_ui_view: Error-prone use of @class in view cart_lines_fee (website_sale_charge_payment_fee.cart_lines_fee): use the hasclass(*classes) function to filter elements by their classes
2020-07-24 14:38:46,033 10307 WARNING oca-ecommerce-unit-tests-11.0-2 odoo.addons.base.ir.ir_ui_view: Error-prone use of @class in view cart_lines_fee (website_sale_charge_payment_fee.cart_lines_fee): use the hasclass(*classes) function to filter elements by their classes
```

File: website_sale_charge_payment_fee/__manifest__.py (hunk 1 above)
Comment (aparragithub) [reply website_sale_charge_payment_fee/__manifest__.py]: Done

File: website_sale_charge_payment_fee/templates/website_sale.xml (hunk 2 above)
Comment (aparragithub) [reply website_sale_charge_payment_fee/templates/website_sale.xml]: Done

**Pull request closed without merging**